app/server/lib/requestUtils.ts

```diff
@@ -40,7 +40,7 @@ export function adaptServerUrl(url: URL, req: RequestWithOrg): void {
   const reqBaseDomain = parseSubdomain(req.hostname).base;
 
   if (process.env.GRIST_SERVE_SAME_ORIGIN === 'true' || req.isCustomHost) {
-    url.hostname = req.hostname;
+    url.host = req.headers.host || req.hostname;
```

Comment:
Can you tell me what this was changed?

Reply:
I need the port for the tests to pass, and in order to take into account environments that use a different port than 80 or 443. The hostname does not include this information.

The explaination is given in the PR description:
> Also a technical note: I had to change the way requestUtils deduce the host of the home, so it takes into account the port. Otherwise [this test failed](https://github.com/gristlabs/grist-core/blob/e9b5b98bcb35e1c9ae9e82eac6e7e8886891cfb4/test/nbrowser/AdminPanel.ts#L337-L343) with my change, as I now specify the absolute URL for serving static resources (needed for specifying the opengraph icon).

I admit it is not the better place for that.
Do you feel this kind of note should rather appear as a comment in the PR?